aws/signer/v4/v4.go

```diff
@@ -230,6 +238,15 @@ func buildAuthorizationHeader(credentialStr, signedHeadersStr, signingSignature
 	return parts.String()
 }
 
+func (v4 Signer) getLogger(ctx context.Context) (logger logging.Logger) {
+	if v4.Logger == nil {
+		logger = logging.Noop{}
```

Comment:
We need to standardize on our nomenclature for no-op. e.g. `smithyhttp.NopClient` vs `logging.Noop`.

Created https://github.com/aws/aws-sdk-go-v2/issues/878